File: llm_generated_code/ChatGPT/Python/Level1/Q23.py
def min_trips(sequence):
    trips = 0
    i = 0
    while i < len(sequence):
        count = 1
        while i + 1 < len(sequence) and sequence[i] == sequence[i + 1]:
            count += 1
            i += 1
        trips += (count + 4) // 5
        i += 1
    return trips

sequence = input()
result = min_trips(sequence)
print(result)

File: llm_generated_code/ChatGPT/CPP/Level1/Q23.cpp
#include <iostream>
#include <string>
using namespace std;

int main() {
    string s;
    cin >> s;
    
    int trips = 0;
    int count = 1;

    for (size_t i = 1; i <= s.length(); i++) {
        if (i < s.length() && s[i] == s[i - 1]) {
            count++;
        } else {
            trips += (count + 4) / 5;
            count = 1;
        }
    }
    
    cout << trips << endl;
    return 0;
}
